aspnet/fundamentals/localization.rst

```diff
@@ -0,0 +1,337 @@
+
+.. code-block:: c#
+	:emphasize-lines: 3-11,21-24
+
+	public class MyRequestCultureProvider : IRequestCultureProvider
+	{
+		public Task<RequestCulture> DetermineRequestCulture(HttpContext httpContext)
+		{
+			// Replace the following with your own logic to determine what
+			// culture should be used. For example, you could read the culture
+			// information from a database keyed by the logged-in user.
+			var culture = new CultureInfo("es-ES");
+			var uiCulture = new CultureInfo("en-ES");
+			return Task.FromResult(new RequestCulture(culture, uiCulture));
+		}
+	}
+
+	public class Startup
+	{
+		public void Configure(IApplicationBuilder app, IStringLocalizer<Startup> SR)
+		{
+			// Add the platform handler to the request pipeline.
+			app.UseIISPlatformHandler();
+
+			var options = new RequestLocalizationOptions();
+			options.RequestCultureProviders.Clear();
+			options.RequestCultureProviders.Add(new MyRequestCultureProvider() );
+			app.UseRequestLocalization(options);
+
+			...
+		}
+	}
+
+Accessing localized content with IStringLocalizer
+-------------------------------------------------
+
+Once the request localization middleware establishes the current culture, the ``IStringLocalizer`` service then provides access to localized content based on the current culture. To enable support for these localization services, you call the ``IServiceCollection.AddLocalization`` extension method specifying a setup action to configure the service. The most common way to call ``IServiceCollection.AddLocalization`` is as follows where the ``ResourcePath`` specifies the path where the localized resources are located. In this example, the resource files reside in a directory named `My/Resources` relative to the application root.
+
+.. code-block:: c#
+	:emphasize-lines: 3
+
+	public void ConfigureServices(IServiceCollection services)
+	{
+		services.AddLocalization(options => options.ResourcesPath = "My/Resources");
+	}
+
+The following screen shot shows the resx files stored in the "My/Resources" directory in a sample project.
+
+.. image:: localization/_static/resx-files.png
+
+As you can see from the figure, your resx files must be named according to a specific pattern:
+
+::
+
+	<type>.<locale>.resx
+
+
+
+
+
+Implementing a custom IStringLocalizer
+--------------------------------------
+
+The default implementations of the localization services is based on `System.Resources.ResourceManager`, which supports accessing localized content in satellite assemblies based on resx files. You can alternatively provide your own implementations for accessing localized content from different sources, like form a database. In this section, we'll take a look at how to implement your own custom ``IStringLocalizer``.
```

Comment:
The fact that the localization services are based on ResourceManager should be described in the "Accessing localized content with IStringLocalizer" section.